ObjectiveGit/GTRepository+Stashing.m

```diff
@@ -68,29 +68,32 @@ - (BOOL)applyStashAtIndex:(NSUInteger)index flags:(GTRepositoryStashApplyFlag)fl
 		stash_options.progress_payload = (__bridge void *)progressBlock;
 	}
 
-	int gitError = git_stash_apply(self.git_repository, index, &stash_options);
-	if (gitError != GIT_OK) {
-		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Stash apply failed" failureReason:@"The stash at index %ld couldn't be applied.", (unsigned long)index];
-		return NO;
+	if (options != nil) {
+		stash_options.checkout_options = *options.git_checkoutOptions;
+	}
+
+	if (pop == NO) {
```

Comment:
This combining of the methods feels off to me since they're pretty different things.